examples/inspection/plot_model_inspection_cross_validation.py

```diff
@@ -0,0 +1,229 @@
+Inspect and analyze a linear model within cross-validation
+==========================================================
+
+Evaluating a predictive linear model involves :ref:`cross-validation
+<cross_validation>`. This example:
+
+* details how to interpret the results from a cross-validation framework;
+* highlights how to inspect the internals of a model when using the
+  cross-validation framework.
+"""
+
+# Authors: Guillaume Lemaitre <g.lemaitre58@gmail.com>
+# License: BSD 3 clause
+
+# %%
+# Dataset
+# -------
+#
+# We will use the :ref:`california_housing_dataset` where the goal is to
+# predict the average house value in a neighborhood. From the start, we will
+# split our data into two sets: a set that we will use for all our experiments
+# and a set that we will leave out for further confirmation.
+
+from sklearn.datasets import fetch_california_housing
+from sklearn.model_selection import train_test_split
+
+X, y = fetch_california_housing(as_frame=True, return_X_y=True)
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
+
+# %%
+# Predictive model
+# ----------------
+#
+# In this example, we will use a linear model which should be a good baseline:
+# a :ref:`ridge model <ridge_regression>`. A ridge model enforces a L2 penalty
+# on the coefficients. Thus, the penalty parameter `alpha` has to be tuned.
+# More importantly, this parameter needs to be tuned for our specific problem:
+# tuning on another dataset does not ensure an optimal parameter value for the
+# current dataset.
+#
+# Here, we define a machine learning pipeline `model` made of a preprocessing
+# stage to :ref:`standardize <preprocessing_scaler>` the data such that the
+# regularization strength is applied homogeneously on each coefficient; followed
+# by a :class:`~sklearn.linear_model.RidgeCV` that will perform an internal
+# leave-one-out cross-validation to select the best parameter `alpha`.
+
+import numpy as np
+from sklearn.linear_model import RidgeCV
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+
+alphas = np.logspace(-1, 2.5, num=50) # candidates for the parameter `alpha`
+model = Pipeline(
+    [
+        ("scaler", StandardScaler()),
+        ("ridge_regressor", RidgeCV(alphas=alphas)),
+    ]
+)
+
+# %%
+# Cross-validation framework
+# --------------------------
+#
+# Before applying such a predictive model on new data, as is often done in
+# production, it is a good practice and generally advisable to perform an
+# unbiased evaluation of the performance of the model.
+#
+# Cross-validation should be used to make this analysis. First, it allows us to
+# quantify the variance of the model performance. A large variance of the score
+# metric will indicate that we cannot trust the reported performance nor try to
+# interpret findings built on internal model's parameters. One possible cause
+# of large variations are small sample sizes.
```

Comment:
With perfect iid data, I can trust the variance / CI interval of a score, which are usually just a mean over the data.
A more important reason, IMO, is violation of iid.